docs/EXT_MANAGER.md

```diff
@@ -0,0 +1,107 @@
+|--------------------|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| whitelistClasses   | String[]         | It contains the list of classes which have common implementation and can be exposed by parent classloader (Naksha service) to child classloader (Extension Jars) during runtime. Default values are "java.*", "javax.*", "com.here.*", "jdk.internal.reflect.*", "com.sun.*", "org.w3c.dom.*", "sun.misc.*". |
+| intervalms         | Long             | Value in Milliseconds, It is being used to create Extension config expiry when the previously loaded config gets expired and so after that, system should make another attempt to refresh the extension config. i.e expiry = System.currentMillis + intervalms                                               |
+| extensionsRootPath | String           | It represents the extension root path where all extension jars are placed. (currently supports only AWS s3 path)                                                                                                                                                                                             |
+
+
+## S3 Folder Structure
+
+Extension manager expects all extension should have a folder under configured extension root path. All Extension should have an ID and it's folder should be created with the same ID. Every extension folder should contain below files:-
+
+- jar - Extension fat jar
+- latest-{ENV}.txt - Text file which contains jar version. This value is being used in determining extension configuration json file. Here {ENV} is the environment name the Naksha service is running against.
+- {ExtensionID}-{VERSION}-{ENV}.json - It is a json file, which contains extension specific configuration. {ExtensionID} is the ID of Extension. {VERSION} is resolved value determined from latest-{ENV}.txt. And {ENV} is the environment name. This file should have below attributes 
+  - id - String representing Extension ID
+  - url - Complete path of jar file (In case of S3, It should be S3 URI of extension jar)
+  - version - Extension version being deployed.
+  - initClassName - Full class name. If configured the Extension manager will load this class and instantiate it.   
+  - properties - Extension specific custom/private properties which is required specifically for that extension. For example, remote URL or Database details. 
+  - type - XYZFeature type. Default value is "Extension".
+
+Below is the sample structure of extension directory. where Root path is `s3://naksha-pvt-releases/extensions/`. And Extension `foo` is deployed on `dev` environment. 
+
+```text
+S3 folder structure:
+  naksha-pvt-releases
```

Comment:
@rlakde , @ashrafsyed - I would like you to confirm if this S3 setup aligns with your individual tests that you did?

Reply:
@hirenkp2000 Yes we did testing with similar S3 setup manually. The "properties" field in {ExtensionID}-{VERSION}-{ENV}.json file is new to me. I never had that in our file. Can you clarify in what cases it is required? Some example specific  to Naksha?
@ashrafsyed Please confirm from Pipeline perspective.

Reply:
Thanks for confirming, @rlakde . Yes, "properties" was not applicable for V&E extension (for now). But it can be helpful for a scenario (in future), where Extension needs some Remote URL or Database credentials as part of one-time initialization, then this is the object by which the information can be provided to Extension.
It is a one-time input per Extension initialization and not per API request.

Reply:
Thanks. Looks good from my side.